Flex (#760)
shopware/docs: guides/installation/flex.md

```

This will create a new project in the `<project-name>` directory. The `dev-flex` version constraint will install the latest version of the Shopware. This constraint `dev-flex` will be removed when the template is generally available.

The template contains all Shopware bundles like `shopware/administration`, `shopware/storefront`, `shopware/elasticsearch`. If you don't need one of them, you can just uninstall it with:
`composer remove shopware/<bundle-name>`.

As next, you have to adjust the generated `.env` file and run following command:

```bash
bin/console system:install --basic-setup
```

This will install Shopware and create a default sales channel with a default admin user named `admin` with password `shopware`. Change these credentials after the installation.

### Optional packages

The template is very small and does not contain any dev-tooling or integrations like PaaS or Fastly. You can easily add them to your project with the following commands:

```bash
# Install Symfony Default Profiler
composer req --dev profiler

# Install PaaS integration
composer req paas

# Install Fastly integration
composer req fastly
```

> ## How to migrate to Symfony Flex?

Reviewer: It's called migration but it does not cover the common case of having extensions installed

Author: it stays same

The template contains all Shopware bundles like `shopware/administration`, `shopware/storefront`, `shopware/elasticsearch`. If you don't need one of them, you can uninstall it with:
`composer remove shopware/<bundle-name>`.

## Installation

### Local installation

You have to adjust the generated `.env` file and run the following command:

```bash
bin/console system:install --basic-setup
```

This installs Shopware and creates a default sales channel with default admin credentials (user name, `admin` and password, `shopware`). Change these credentials after the installation.

### Installation with Docker and local PHP using Symfony CLI

The Symfony CLI is a developer tool that helps to build, run, and manage your Symfony applications directly from the terminal. The services will run in Docker containers and the application will run locally.

* Requisites: 

1. Install [Symfony CLI](https://symfony.com/download), if you don't have one. 

2. As Symfony CLI uses local PHP, ensure to have PHP installed. Below are the commands to install it:

{% tabs %}
{% tab title="Ubuntu based Linux" %}

You need to add a new Software Repository to your system to have the latest PHP version.

> ```bash
> sudo add-apt-repository ppa:ondrej/php
> 
> sudo apt-get install -y php8.1-fpm php8.1-mysql php8.1-curl php8.1-gd php8.1-xml php8.1-zip php8.1-opcache php8.1-mbstring php8.1-intl php8.1-cli

Reviewer 2: ```suggestion
```bash
sudo add-apt-repository ppa:ondrej/php

sudo apt-get install -y php8.1-fpm php8.1-mysql php8.1-curl php8.1-gd php8.1-xml php8.1-zip php8.1-opcache php8.1-mbstring php8.1-intl php8.1-cli
```
```

> ```bash
> sudo add-apt-repository ppa:ondrej/php
> 
> sudo apt-get install -y php8.1-fpm php8.1-mysql php8.1-curl php8.1-gd php8.1-xml php8.1-zip php8.1-opcache php8.1-mbstring php8.1-intl php8.1-cli
> 

Reviewer 2: ```suggestion
```bash
sudo add-apt-repository ppa:ondrej/php

sudo apt-get install -y php8.1-fpm php8.1-mysql php8.1-curl php8.1-gd php8.1-xml php8.1-zip php8.1-opcache php8.1-mbstring php8.1-intl php8.1-cli
```

```

### Optional packages

The template is small and does not contain any dev-tooling or integrations like PaaS or Fastly. You can easily add them to your project with the following commands:

```bash
# Install Symfony Default Profiler
composer req --dev profiler

# Install PaaS integration
composer req paas

# Install Fastly integration
composer req fastly
```

> #### Adding Shopware Packagist
> 
> Using Shopware Packagist you can manage all your Shopware Store plugins directly in the composer.json aswell. See this blog post for [more information](https://www.shopware.com/en/news/using-composer-for-plugin-installation-in-shopware/).

Reviewer 2: ```suggestion
#### Add Shopware Packagist

Using Shopware Packagist you can manage all your Shopware Store plugins directly in the composer.json. Refer to ["Using composer for plugin installation in Shopware"](https://www.shopware.com/en/news/using-composer-for-plugin-installation-in-shopware/) blog post for detailed information.
```
